decompressed/base/lib/mount_modroot/05_transfer_basefiles

```diff
@@ -3,7 +3,31 @@
 upperdir=$1
 overlay_bank=/overlay/bank_2
 
-preserve_list="/etc/init.d/rootdevice /etc/rc.d/S94rootdevice /usr/sbin/random_seed /sbin/mount_root-mod /lib/mount_modroot/04_move_overlay /lib/mount_modroot/05_transfer_basefiles"
+#use the www folder as check to know that we should populate the modoverlay (mtd3) and cleanup real overlay (mtd2)
+if [ -d $overlay_bank/www ]; then
+
+  #iterate all bank_2 folders and file to merge content with the one on modoverlay
+  for file in $(find "$overlay_bank/"*/ -xdev | cut -d '/' -f4-); do
+    #create folder if not exist on destination
+    if [[ -d "$overlay_bank/$file" && ! -d "$upperdir/$file" ]]; then
+      mkdir "$upperdir/$file"
+      continue
+    fi
+
+    #if not a dir move the file (and symlink)
+    [ ! -d "$overlay_bank/$file" ] && mv -f "$overlay_bank/$file" "$upperdir/$file"
+
+  done
+
+  #at this point we should remove all empty folders in bank_2 overlay, except the modoverlay mountpoint
+  for f in "$overlay_bank/"*; do
+    [ "$f" != "$overlay_bank$upperdir" ] && rm -rf "$f"
+  done
+
+fi
+
```

Comment:
Non sono sicuro possa rappresentare un problema ma per sicurezza è meglio se metti il codice delle due operazioni in una funzione e poi come ultima riga del file la chiami. In questo modo arrivi qui a metà essendo sicuro che il parsing di questo file sia già stato fatto tutto in precedenza. Non ho mai indagato a fondo come funzioni l'interprete di sh in questi casi ma posso dirti che ad esempio se questo file modifica se stesso aggiungendo comandi in fondo questi verrebbero eseguiti, quindi di sicuro lui non fa il parsing solo prima di eseguirlo.

Reply:
mindfuck, next level.

comunque fatto.. e ora che reboot sia.